public/app/plugins/panel/heatmap/utils.ts

```diff
@@ -423,11 +423,15 @@ export function prepConfig(opts: PrepConfigOpts) {
           : dataRef.current?.xLayout === HeatmapCellLayout.ge
             ? 1
             : 0,
-      yAlign: ((dataRef.current?.yLayout === HeatmapCellLayout.le
-        ? -1
-        : dataRef.current?.yLayout === HeatmapCellLayout.ge
-          ? 1
-          : 0) * (yAxisReverse ? -1 : 1)) as -1 | 0 | 1,
+      yAlign: (() => {
+        const yAlign =
+          dataRef.current?.yLayout === HeatmapCellLayout.le
+            ? -1
+            : dataRef.current?.yLayout === HeatmapCellLayout.ge
+              ? 1
+              : 0;
+        return yAxisReverse ? (yAlign === -1 ? 1 : yAlign === 1 ? -1 : 0) : yAlign;
+      })(),
```

Comment:
TypeScript isn't smart enough to know that `-1 * (-1, 0, 1)` is still `-1 | 0 | 1`, so the simplest way to get this to not yell is to just avoid doing math and use another ternary.